main.tf

```diff
@@ -144,6 +143,17 @@ resource "aws_elasticache_replication_group" "default" {
   final_snapshot_identifier  = var.final_snapshot_identifier
   apply_immediately          = var.apply_immediately
 
+  dynamic "log_delivery_configuration" {
+    for_each = var.log_delivery_configuration
```

Comment:
should we convert it to a map (in locals) to prevent updating the whole list if any item changes?

Reply:
Only two items can be used here for each log_type. We could turn it into a map but i think it would be overkill.